packages/@react-aria/tag/src/useTag.ts

```diff
@@ -56,14 +54,14 @@ export function useTag(props: TagProps<any>, state: GridState<any, any>): TagAri
     focusMode: 'cell'
   }, state, tagRef);
 
-  function onKeyDown(e: KeyboardEvent<Element>) {
+  function onKeyDown(e: KeyboardEvent) {
     if (e.key === 'Delete' || e.key === 'Backspace' || e.key === ' ') {
-      onRemove(children, e);
+      onRemove(gridCellProps['data-key']);
```

Comment:
You should use `item.key` here instead. Data attributes will be converted to strings, but keys can also be numbers. Same below.